File: dashboards/laravel/views/guardrails.php
<?php
/**
 * Guardrails page — agent controls, interventions, and anomaly rules.
 * Expects: $interventionItems (array of interventions), $anomalyRuleItems (array of anomaly rules)
 */
include_once __DIR__ . '/components/table.php';

$headers = ['Agent', 'Reason', 'Status', 'Time', 'Actions'];
$rows = [];
foreach ($interventionItems as $i) {
    $id     = Helpers::e($i['id'] ?? '');
    $agent  = '<strong>' . Helpers::e($i['agent'] ?? $i['agentName'] ?? '-') . '</strong>';
    $reason = Helpers::e($i['reason'] ?? '-');
    $status = Helpers::statusBadge($i['status'] ?? 'active');
    $time   = Helpers::timeAgo($i['created_at'] ?? $i['timestamp'] ?? '-');

    $actions = '';

    $pauseForm  = '<form method="post" action="/guardrails" style="display:inline">';
    $pauseForm .= '<input type="hidden" name="_action" value="pause">';
    $pauseForm .= '<input type="hidden" name="id" value="' . $id . '">';
    $pauseForm .= '<button type="submit" class="btn btn-sm btn-warning">Pause</button>';
    $pauseForm .= '</form> ';

    $resumeForm  = '<form method="post" action="/guardrails" style="display:inline">';
    $resumeForm .= '<input type="hidden" name="_action" value="resume">';
    $resumeForm .= '<input type="hidden" name="id" value="' . $id . '">';
    $resumeForm .= '<button type="submit" class="btn btn-sm btn-primary">Resume</button>';
    $resumeForm .= '</form> ';

    $killForm  = '<form method="post" action="/guardrails" style="display:inline">';
    $killForm .= '<input type="hidden" name="_action" value="kill">';
    $killForm .= '<input type="hidden" name="id" value="' . $id . '">';
    $killForm .= '<button type="submit" class="btn btn-sm btn-danger" onclick="return confirm(\'Kill this guardrail?\')">Kill</button>';
    $killForm .= '</form>';

    $actions = $pauseForm . $resumeForm . $killForm;

    $rows[] = [$agent, $reason, $status, $time, $actions];
}
?>

<div class="card">
    <h3>Interventions</h3>
    <?= renderTable($headers, $rows) ?>
</div>

<div class="card">
    <h3>Create Anomaly Rule</h3>
    <form method="post" action="/guardrails" class="inline-form">
        <input type="hidden" name="_action" value="create_rule">
        <div class="form-group" style="flex:1;min-width:140px;margin-bottom:0">
            <input type="text" name="name" placeholder="Rule name" required>
        </div>
        <div class="form-group" style="flex:1;min-width:140px;margin-bottom:0">
            <input type="text" name="condition" placeholder="Condition" required>
        </div>
        <div class="form-group" style="min-width:120px;margin-bottom:0">
            <select name="action">
                <option value="alert">Alert</option>
                <option value="block">Block</option>
                <option value="pause">Pause</option>
            </select>
        </div>
        <div class="form-group" style="min-width:100px;margin-bottom:0">
            <input type="number" name="threshold" placeholder="Threshold" min="0" value="0">
        </div>
        <button type="submit" class="btn btn-primary">Create</button>
    </form>
</div>

<?php
$aHeaders = ['Name', 'Condition', 'Action', 'Threshold', 'Actions'];
$aRows = [];
foreach ($anomalyRuleItems as $r) {
    $id        = Helpers::e($r['id'] ?? '');
    $name      = '<strong>' . Helpers::e($r['name'] ?? '-') . '</strong>';
    $condition = Helpers::e($r['condition'] ?? '-');
    $action    = Helpers::statusBadge($r['action'] ?? 'alert');
    $threshold = Helpers::e((string)($r['threshold'] ?? '0'));

    $deleteForm  = '<form method="post" action="/guardrails" style="display:inline">';
    $deleteForm .= '<input type="hidden" name="_action" value="delete_rule">';
    $deleteForm .= '<input type="hidden" name="id" value="' . $id . '">';
    $deleteForm .= '<button type="submit" class="btn btn-sm btn-danger" onclick="return confirm(\'Delete this rule?\')">Delete</button>';
    $deleteForm .= '</form>';

    $aRows[] = [$name, $condition, $action, $threshold, $deleteForm];
}
?>

<div class="card">
    <h3>Anomaly Rules</h3>
    <?= renderTable($aHeaders, $aRows) ?>
</div>

File: dashboards/go/handlers/guardrails.go
package handlers

import (
	"agenticmail-dashboard/services"
	"agenticmail-dashboard/templates"
	"fmt"
	"net/http"
)

// HandleGuardrails handles the guardrails page (GET) with interventions and rules,
// and POST actions for pause, resume, kill, create_rule, and delete_rule.
func HandleGuardrails(w http.ResponseWriter, r *http.Request) {
	s := services.GetSession(r)

	if r.Method == "POST" {
		r.ParseForm()
		action := r.FormValue("action")
		switch action {
		case "pause":
			services.APICall("/engine/guardrails/pause/"+r.FormValue("id"), "POST", s.Token, nil)
		case "resume":
			services.APICall("/engine/guardrails/resume/"+r.FormValue("id"), "POST", s.Token, nil)
		case "kill":
			services.APICall("/engine/guardrails/kill/"+r.FormValue("id"), "POST", s.Token, nil)
		case "create_rule":
			services.APICall("/engine/anomaly-rules", "POST", s.Token, map[string]string{
				"name": r.FormValue("name"), "condition": r.FormValue("condition"),
				"action": r.FormValue("rule_action"),
			})
		case "delete_rule":
			services.APICall("/engine/anomaly-rules/"+r.FormValue("id"), "DELETE", s.Token, nil)
		}
		http.Redirect(w, r, "/guardrails", http.StatusFound)
		return
	}

	interventions, _ := services.APICall("/engine/guardrails/interventions", "GET", s.Token, nil)
	rulesData, _ := services.APICall("/engine/anomaly-rules", "GET", s.Token, nil)

	var interventionsHTML string
	if iList, ok := interventions["interventions"].([]interface{}); ok && len(iList) > 0 {
		rows := ""
		for _, iv := range iList {
			i := iv.(map[string]interface{})
			actions := fmt.Sprintf(`<form method="POST" action="/guardrails" style="display:inline-flex;gap:4px"><input type="hidden" name="id" value="%s">`, templates.Esc(i["id"]))
			status := templates.StrVal(i, "status")
			if status == "active" {
				actions += `<button class="btn btn-sm" type="submit" name="action" value="pause">Pause</button>`
				actions += `<button class="btn btn-sm btn-d" type="submit" name="action" value="kill">Kill</button>`
			} else if status == "paused" {
				actions += `<button class="btn btn-sm" type="submit" name="action" value="resume">Resume</button>`
				actions += `<button class="btn btn-sm btn-d" type="submit" name="action" value="kill">Kill</button>`
			}
			actions += `</form>`
			rows += fmt.Sprintf(`<tr><td style="font-weight:600">%s</td><td style="color:var(--dim)">%s</td><td>%s</td><td style="font-size:12px;color:var(--muted)">%s</td><td>%s</td></tr>`,
				templates.Esc(i["agent"]), templates.Esc(i["reason"]), templates.Badge(status), templates.Esc(i["timestamp"]), actions)
		}
		interventionsHTML = `<table><thead><tr><th>Agent</th><th>Reason</th><th>Status</th><th>Time</th><th></th></tr></thead><tbody>` + rows + `</tbody></table>`
	} else {
		interventionsHTML = `<div class="empty"><div class="empty-i">🛡️</div>No active interventions</div>`
	}

	var rulesHTML string
	if ruleList, ok := rulesData["rules"].([]interface{}); ok && len(ruleList) > 0 {
		rows := ""
		for _, rl := range ruleList {
			ru := rl.(map[string]interface{})
			rows += fmt.Sprintf(`<tr><td style="font-weight:600">%s</td><td style="color:var(--dim)">%s</td><td>%s</td><td><form method="POST" action="/guardrails" style="display:inline"><input type="hidden" name="action" value="delete_rule"><input type="hidden" name="id" value="%s"><button class="btn btn-sm btn-d" type="submit">Delete</button></form></td></tr>`,
				templates.Esc(ru["name"]), templates.Esc(ru["condition"]), templates.Badge(templates.StrVal(ru, "action")), templates.Esc(ru["id"]))
		}
		rulesHTML = `<table><thead><tr><th>Name</th><th>Condition</th><th>Action</th><th></th></tr></thead><tbody>` + rows + `</tbody></table>`
	} else {
		rulesHTML = `<div class="empty"><div class="empty-i">📏</div>No anomaly rules yet</div>`
	}

	content := fmt.Sprintf(`<h2 class="t">Guardrails</h2><p class="desc">Monitor and control agent behavior</p>
<div class="card" style="margin-bottom:16px"><div class="ct">Create Anomaly Rule</div>
<form method="POST" action="/guardrails" style="display:flex;gap:10px;align-items:end">
<input type="hidden" name="action" value="create_rule">
<div class="fg" style="flex:1;margin:0"><label class="fl">Name</label><input class="input" name="name" required placeholder="e.g. Rate limit exceeded"></div>
<div class="fg" style="flex:1;margin:0"><label class="fl">Condition</label><input class="input" name="condition" required placeholder="e.g. messages > 100/min"></div>
<div class="fg" style="margin:0"><label class="fl">Action</label><select class="input" name="rule_action"><option>pause</option><option>alert</option><option>kill</option></select></div>
<button class="btn btn-p" type="submit">Create</button></form></div>
<div class="card" style="margin-bottom:16px"><div class="ct">Active Interventions</div>%s</div>
<div class="card"><div class="ct">Anomaly Rules</div>%s</div>`, interventionsHTML, rulesHTML)

	w.Header().Set("Content-Type", "text/html")
	fmt.Fprint(w, templates.Layout("guardrails", s.User, content))
}
